### PR TITLE
[Technical debt]  LPS-171817 Fix some localization issues of View Builder

### PR DESCRIPTION
**Pull request by alinedoleron**
[LPS-171817](https://issues.liferay.com/browse/LPS-171817)
This is a technical debt that fixes some localization issues of View Builder

### CODE REVIEW

**Comment by alinedoleron** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by alinedoleron** [comment]
ci:test:relevant

**Review by danvb112 — CHANGES_REQUESTED** [review]
Hey @alinedoleron, I haven't finished the review yet, but I have a few suggestions about it. :smile:

File: modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/components/BuilderScreen/BuilderScreen.tsx [hunk 1]
```diff
@@ -151,9 +155,10 @@ export function BuilderScreen({
 												  )
 											: filter
 											? viewColumn?.objectFieldBusinessType
-											: viewColumn?.label[
-													defaultLanguageId
-											  ]
+											: getLocalizableLabel(
+													viewColumn?.label,
+													creationLanguageId!
+											  )
```
Comment (danvb112) [inline modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/components/BuilderScreen/BuilderScreen.tsx:161]: @alinedoleron we can use the `viewColumn.objectFieldName` as the `fallback` here.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectView/ViewBuilderScreen/ViewBuilderScreen.tsx [hunk 2]
```diff
@@ -49,12 +51,14 @@ const ViewBuilderScreen: React.FC<{}> = () => {
 		const parentWindow = Liferay.Util.getOpener();
 
 		parentWindow.Liferay.fire('openModalAddColumns', {
-			getName: ({label}: ObjectField) => label[defaultLanguageId],
+			getName: ({label}: ObjectField) =>
+				getLocalizableLabel(label, creationLanguageId),
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectView/ViewBuilderScreen/ViewBuilderScreen.tsx:55]: @alinedoleron we can extract the `name` from `ObjectField` and use as the `fallback` here.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectView/index.tsx [hunk 3]
```diff
@@ -79,6 +79,10 @@ const CustomView: React.FC<React.HTMLAttributes<HTMLElement>> = () => {
 				objectDefinitionExternalReferenceCode
 			);
 
+			const objectDefinition = await API.getObjectDefinitionById(
+				objectDefinitionId
+			);
+
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectView/index.tsx:85]: Hey @alinedoleron, As of this [Epic here](https://issues.liferay.com/browse/LPS-164800), we are now using the ERC endpoints to do all GET on Object Definitions, so could you change this endpoint to `getObject Definition ExternalReferenceCode()`?

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectView/objectViewContext.tsx [hunk 4]
```diff
@@ -253,9 +265,10 @@ const viewReducer = (state: TState, action: TAction) => {
 						if (objectField.name === sortColumn.objectFieldName) {
 							newObjectViewSortColumns.push({
 								...sortColumn,
-								fieldLabel: objectField.label[
-									defaultLanguageId
-								] as string,
+								fieldLabel: getLocalizableLabel(
+									objectField.label,
+									creationLanguageId
+								),
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectView/objectViewContext.tsx:271]: Hey @alinedoleron, you can use the `objectField.name` as the `fallback` of the function

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectView/objectViewContext.tsx [hunk 5]
```diff
@@ -313,7 +326,10 @@ const viewReducer = (state: TState, action: TAction) => {
 						...filterColumn,
 						definition,
 						fieldLabel: objectField
-							? objectField.label[defaultLanguageId]
+							? getLocalizableLabel(
+									objectField.label,
+									creationLanguageId
+							  )
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectView/objectViewContext.tsx:332]: Hey @alinedoleron, you can use the `objectField.name` as the `fallback` of the function

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectView/objectViewContext.tsx [hunk 6]
```diff
@@ -351,7 +378,10 @@ const viewReducer = (state: TState, action: TAction) => {
 					return {
 						...item,
 						defaultSort: defaultSortColumn ? true : false,
-						fieldLabel: item.label[defaultLanguageId],
+						fieldLabel: getLocalizableLabel(
+							item.label,
+							creationLanguageId
+						),
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectView/objectViewContext.tsx:384]: Hey @alinedoleron, you can use the `item.name` as the `fallback` of the function.

**Review by danvb112 — CHANGES_REQUESTED** [review]
Hey @alinedoleron, now I'm done reviewing your PR, I have a few suggestions, most of which are related to normally using field names as a fallback to the new `getLocalizableLabel` function.

**Comment by alinedoleron** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:relevant

**Review by danvb112 — APPROVED** [review]
LGTM! Great job @alinedoleron! :tada:

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by alinedoleron** [comment]
Manually forward to https://github.com/brianchandotcom/liferay-portal/pull/128765

**Pull request closed without merging**